File: 13-roman-to-integer/13-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        mp={'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        ans=0
        
        for i in range(0,len(s)-1):
            if s[i] == 'I':
                if s[i+1] == 'V' or s[i+1] == 'X':
                    ans+=-1
                else:
                    ans+=1
            elif s[i] == 'X':
                if s[i+1] == 'L' or s[i+1] == 'C':
                    ans+=-10
                else:
                    ans+=10
            elif s[i] == 'C':
                if s[i+1] == 'D' or s[i+1] == 'M':
                    ans+=-100
                else:
                    ans+=100
            else:
                ans+=mp[s[i]]
        ans+=mp[s[len(s)-1]]
        return ans
        

File: 13-roman-to-integer/13-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char,int> mp;
        mp.insert({'I',1});
        mp.insert({'V',5});
        mp.insert({'X',10});
        mp.insert({'L',50});
        mp.insert({'C',100});
        mp.insert({'D',500});
        mp.insert({'M',1000});
        
        int ans=0;
        for(int i=0;i<s.size()-1;i++)
        {
            if(s[i]=='C') 
            {
                if(s[i+1]=='D' || s[i+1]=='M') 
                    ans+=-100;
                else
                    ans+=100;
            }
            else if(s[i]=='X') 
            {
                if(s[i+1]=='L' || s[i+1]=='C')
                    ans+=-10;
                else 
                    ans+=10;
            }
            else if(s[i]=='I') 
            {
                if(s[i+1]=='X' || s[i+1]=='V') 
                    ans+=-1;
                else 
                    ans+=1;
            }
            else 
                ans+=mp[s[i]];
        }
        ans+=mp[s[s.size()-1]];
        return ans;
    }
};